wifi-subsys/components/protocols/test/test_icmp.c

```diff
@@ -13,51 +13,110 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
+#include "esp_log.h"
 #include "unity.h"
 #include "wifi.h"
 #include "storage.h"
 #include "icmp_ping.h"
 #include "freertos/FreeRTOS.h"
 #include "freertos/task.h"
+#include "default_params.h"
+
+esp_err_t set_default_ping_settings(void){
+    ESP_LOGI(__func__, "Default ping settings");
+    esp_err_t err = nvs_set_string(stringlify(PING), stringlify(PING_TO), PING_TO);
+    if (err != ESP_OK)
+    {
+        ESP_LOGE(__func__,
+                "Error: setting ping address to the nvs %s",
+                esp_err_to_name(err));
+    }
+    err = nvs_set_uint32(stringlify(PING), stringlify(PING_INT), PING_INT);
+    if (err != ESP_OK)
+    {
+        ESP_LOGE(__func__,
+                "Error: setting ping interval to the nvs %s",
+                esp_err_to_name(err));
+    }
+    err = nvs_set_uint8(stringlify(PING), stringlify(PING_RET), PING_RET);
+    if (err != ESP_OK)
+    {
+        ESP_LOGE(__func__,
+                "Error: setting ping retries to the nvs %s",
+                esp_err_to_name(err));
+    }
+    err = nvs_set_uint32(stringlify(PING), stringlify(PING_TIMEOUT), PING_TIMEOUT);
+    if (err != ESP_OK)
+    {
+        ESP_LOGE(__func__,
+                "Error: setting ping timeout to the nvs %s",
+                esp_err_to_name(err));
+    }
+     if (err != ESP_OK) {
+        nvs_set_uint8(stringlify(NVS), stringlify(NVS_IS_INIT), 0);
+        return err;
+    }
+    else {
+        nvs_set_uint8(stringlify(NVS), stringlify(NVS_IS_INIT), 1);
+        return ESP_OK;
+    }
+}
 
 TEST_CASE("Wi-Fi initialize", "[network]")
 {
     esp_err_t err;
     nvs_init();
-    set_default_credentials();
-    err = wifi_init();
-
-    if (err != ESP_OK) TEST_FAIL();
+    wifi_init();
+    err = set_default_credentials();
+     if (err != ESP_OK) {
+        TEST_FAIL();
+    };
 }
 
 TEST_CASE("setting ping session", "[network]")
 {
-   vTaskDelay(10000 / portTICK_PERIOD_MS);
-
-   esp_err_t err = initialize_ping();
-   if (err != ESP_OK) {
-       TEST_FAIL();
-   }
+    nvs_init();
+    esp_err_t err = set_default_credentials();
+    if (err != ESP_OK) {
+        TEST_FAIL();
+    };
+    set_default_ping_settings();
+    vTaskDelay(10000 / portTICK_PERIOD_MS);
+    initialize_ping();
 }
 
 void icmp_callback(uint8_t is_connected)
 {
     printf("nuevos parametros %d \n", is_connected);
 }
-
 TEST_CASE("send manual ping", "[network]")
 {
-   manual_ping(icmp_callback);
-//    if (is_connected == 0) {
-//        TEST_FAIL();
-//    }
+    nvs_init();
+     esp_err_t err = set_default_credentials();
+    if (err != ESP_OK) {
+        TEST_FAIL();
+    };
+    manual_ping(icmp_callback);
+}
+TEST_CASE("start ping task", "[network]")
+{
+    nvs_init();
+    esp_err_t err = set_default_credentials();
```

Comment:
change `set_default_credentials` for  `set_default_ping_settings`  i think you don't need this or yes?